src/transformers/models/rag/modeling_tf_rag.py

```diff
@@ -588,23 +603,23 @@ def call(
         )
 
         # aliasing to minimize code changing
-        input_ids=inputs["input_ids"]
-        attention_mask=inputs["attention_mask"]
-        decoder_input_ids=inputs["decoder_input_ids"]
-        decoder_attention_mask=inputs["decoder_attention_mask"]
-        encoder_outputs=inputs["encoder_outputs"]
-        past_key_values=inputs["past_key_values"]
-        doc_scores=inputs["doc_scores"]
-        context_input_ids=inputs["context_input_ids"]
-        context_attention_mask=inputs["context_attention_mask"]
+        input_ids = inputs["input_ids"]
+        attention_mask = inputs["attention_mask"]
+        decoder_input_ids = inputs["decoder_input_ids"]
+        decoder_attention_mask = inputs["decoder_attention_mask"]
+        encoder_outputs = inputs["encoder_outputs"]
+        past_key_values = inputs["past_key_values"]
+        doc_scores = inputs["doc_scores"]
+        context_input_ids = inputs["context_input_ids"]
+        context_attention_mask = inputs["context_attention_mask"]
 
         use_cache = inputs["use_cache"]
         output_attentions = inputs["output_attentions"]
         output_hidden_states = inputs["output_hidden_states"]
         return_dict = inputs["return_dict"]
         n_docs = inputs["n_docs"] if inputs["n_docs"] is not None else self.config.n_docs
```

Comment:
This should be handled in the processing method not here.

Reply:
@jplu Sorry for late reply (just finished `TFRagSequenceForGeneration`). Do I understand correctly that you meant I should "modify" input_processing in `modeling_tf_utils` to handle `n_docs`, `do_marginalize` and `reduce_loss`  ??